### PR TITLE
bpo-46829: Deprecate passing a message into Future.cancel() and Task.cancel()

### PR DESCRIPTION
**Pull request by asvetlov**
https://bugs.python.org/issue46829

### CODE REVIEW

File: Lib/test/test_asyncio/test_taskgroups.py
Comment (gvanrossum) [inline Lib/test/test_asyncio/test_taskgroups.py]: I would just remove the “test” arg to cancel(), here and below in this file. It serves no real purpose.

Comment (asvetlov) [reply Lib/test/test_asyncio/test_taskgroups.py]: Done

File: Lib/test/test_asyncio/test_taskgroups.py
Comment (gvanrossum) [inline Lib/test/test_asyncio/test_taskgroups.py]: I think the same here.

**Comment by gvanrossum** [comment]
Please hold off on merging this until we've come to some kind of agreement on whether to deprecate `.cancel(msg)`. See bpo-46829.

**Comment by gvanrossum** [comment]
Also, *if* we do this we should add deprecation markup to the docs. Search for 'method:: cancel' in asyncio-future.rst and asyncio-task.rst. Also there's a note at the very bottom of asyncio-future.rst about the difference between asyncio.Future.cancel and concurrent.futures.Future.cancel that should be deprecated.

**Comment by asvetlov** [comment]
Forgot about docs, thanks.
Sure, I'll wait for the final decision. 
Just wanted to don't do it at the last moment before the feature freeze.

**Comment by asvetlov** [comment]
I've updated the docs.

**Review by gvanrossum — APPROVED** [review]
I don't know how long to wait for @cjerdonek to respond or agree.

**Review by 1st1 — APPROVED** [review]

**Comment by asvetlov** [comment]
@cjerdonek would you comment on this pull request, please?

**Comment by gvanrossum** [comment]
I'm just going to merge this. @cjerdonek, if you have a convincing argument to keep this, we can always revert this PR.

**Comment by gvanrossum** [comment]
At the core dev sprint we discussed this and decided to revert this. I'm aiming to get the reversal into 3.11.

**Pull request merged**
